nifi-nar-bundles/nifi-standard-services/nifi-hbase_2-client-service-bundle/nifi-hbase_2-client-service/src/main/java/org/apache/nifi/hbase/phoenix/StandardPhoenixDBCPService.java

```diff
@@ -0,0 +1,196 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.hbase.phoenix;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.dbcp.BasicDataSource;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnDisabled;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.dbcp.DBCPService;
+import org.apache.nifi.expression.AttributeExpression;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.phoenix.jdbc.PhoenixDriver;
+
+@Tags({ "phoenix", "DBCP", "hbase" })
+@CapabilityDescription("Create a phoenix DBCP connection pool.")
+public class StandardPhoenixDBCPService extends AbstractControllerService implements DBCPService {
+
+    public static final PropertyDescriptor DATABASE_URL = new PropertyDescriptor.Builder()
+            .name("Database Connection URL")
+            .description(
+                    "A database connection URL used to connect to a database. May contain database system name, host, port, database name and some parameters."
+                            + " The exact syntax of a database connection URL is specified by your DBMS.")
+            .defaultValue(null).addValidator(StandardValidators.NON_EMPTY_VALIDATOR).required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY).build();
+
+    public static final PropertyDescriptor DB_DRIVER_TYPE = new PropertyDescriptor.Builder()
+            .name("Phoenix Driver to use").description("Phoenix Driver to use").defaultValue("Thick")
+            .allowableValues("Thick", "Thin").required(true).addValidator(Validator.VALID).build();
+
+    public static final PropertyDescriptor DB_USER = new PropertyDescriptor.Builder().name("Database User")
+            .description("Database user name").defaultValue(null).addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
Instead of the generic 'database' terms, the documentation could refer to Phoenix / HBase.

Reply:
you mean in the description?, will do that change.